content/docs/command-reference/gc.md

```diff
@@ -44,8 +44,29 @@ If the `--cloud` (`-c`) flag is used, this command deletes unused data from the
 to deleting it from the local DVC cache. To specify a DVC remote to delete from,
 use the `--remote` (`-r`) option.
 
-> ⚠️ Danger: cloud deletion is irreversible unless there is another DVC remote
-> or a manual backup with the same data.
+<admon type="warn">
+
+Cloud deletion is irreversible unless there is another DVC remote or a manual
+backup with the same data.
+
+</admon>
+
+### Cleaning shared cache or shared remote
+
+If a single remote or a single [cache is shared] among different projects and
+those projects have overlapped DVC-tracked files, removing files via `gc` in one
+project may break some links in another project.
+
+To keep overlapping files `dvc gc` has a flag `--projects` (`-p`) that takes
+multiple path to DVC projects. DVC _keeps_ data referenced in those projects.
+Make sure that all the commits (including branches) that are referencing any
+files to be preserved have been pulled before running this command.
+
+For example, if we have `N` projects that have some overlapping files and we'd
+like to remove all data in one of those (let's say we don't need one of the
+projects anymore), we would need first to clone all the remaining `N-1` projects
+(including branches) and pass their path to the `dvc gc -p` command while
+running it from the project we'd like to delete the data from.
```

Comment:
```suggestion
For example, if we have 4 projects with some overlapping files and we'd
like to remove all the data that's only needed for one of them (e.g. when we no longer need that projects),
we would first clone all the other projects and fetch all their branches,
then pass their paths to the `dvc gc -p` command from
the project we want to clear.
```

Reply:
Maybe we can move this to an actual example instead of providing an example in a subsection of the description. Or skip the example.

Reply:
mentioning 4 and not mentioning 3 after has less sense. Then it's better to use `we have a few project` ... (I was trying to make a bit more explicit). Again, not a strong opinion at all ... let's merge all this stuff

Reply:
changed to "several"